File: project-euler.java/src/main/java/euler/solutions/_19.java
package euler.solutions;

import euler.common.BaseSolution;

import java.time.LocalDate;

import static java.time.DayOfWeek.SUNDAY;

public class _19 extends BaseSolution
{
    @Override
    public long solve() {
        int sundays_first = 0;
        for (int year = 1901; year <= 2000; year++) {
            for (int month = 1; month <= 12; month++) {
                if (LocalDate.of(year, month, 1).getDayOfWeek() == SUNDAY) {
                    sundays_first++;
                }
            }
        }
        return sundays_first;
    }
}

File: project-euler.go/solutions/_19/19.go
package _19

import "time"

func Solve() int64 {
	sundays_first := 0
	for year := 1901; year <= 2000; year++ {
		for month := 1; month <= 12; month++ {
			t := time.Date(year, time.Month(month), 1, 0, 0, 0, 0, time.UTC)
			if t.Weekday() == time.Sunday {
				sundays_first++
			}
		}
	}
	return int64(sundays_first)
}
